File: emoji.go
package main

import "os"
import "io"
import "fmt"
import "encoding/json"
import "strings"

// bytes, err := ioutil.ReadAll(os.Stdin)
// if err != nil {
//     fmt.Println("Error when reading stdin", err)
//     return
// }

type Emoji struct {
    Emoji, Description, Category, Unicode_version, Ios_version string
    Aliases, Tags []string
}

func load_emojis() []Emoji {
    stream, err := os.Open("emoji.json")
    if err != nil {
        panic(err)
    }

    decoder := json.NewDecoder(stream)

    for {
        var emojis []Emoji
        if err := decoder.Decode(&emojis); err == io.EOF {
            break
        } else if err != nil {
            fmt.Println("Got error:", err)
        }
        return emojis
    }

    var emojis []Emoji
    return emojis

}

func getData() string{
    return "hello :+1: world :fr:"
}

func main() {

    data := getData()

    emojis := load_emojis()
    for _, emoji := range emojis {
        for _, alias := range emoji.Aliases {
            data = strings.Replace(data, ":" + alias + ":", emoji.Emoji, -1)
        }
    }

    print(data)

}

File: emoji.py
# -*- encoding: utf-8 -*-

import sys
import codecs
import json

def load_emojis():
    with open('emoji.json', encoding='utf-8') as fp:
        return json.load(fp)

def print_usage():
    print("Just pipe me!")
    print('\necho hello world :tada: | python emoji.py')

def main():
    return print(unicode('😀'))
    if not sys.stdin.isatty():
        data = sys.stdin.read()
    else:
        print_usage()
        exit()

    emojis = load_emojis()

    for emoji in emojis:
        if 'emoji' not in emoji.keys():
            continue
        for alias in emoji['aliases']:
            data = data.replace(':' + alias + ':', emoji['emoji'])

    with open(sys.stdout.fileno(), 'w', encoding='utf-8') as fp:
        print(data, file=fp)
    print('😀')

if __name__ == '__main__':
    main()
